File: DebriefRepository/src/main/java/org/example/debriefrepository/entity/Debrief.java
package org.example.debriefrepository.entity;

import jakarta.persistence.*;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

import java.time.ZonedDateTime;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@Entity
@Table(name = "debrief")
public class Debrief extends BaseEntity {

    @Column(name = "date", nullable = false)
    private ZonedDateTime date;

    @Column(name = "title", nullable = false)
    private String title;

    @Column(name = "labels", nullable = false)
    private String labels;

    @OneToMany(mappedBy = "debrief", cascade = CascadeType.ALL,
            fetch = FetchType.LAZY, orphanRemoval = true)
    private List<ContentItem> contentItems = new ArrayList<>();

    @OneToMany(mappedBy = "debrief", cascade = CascadeType.ALL,
            fetch = FetchType.LAZY, orphanRemoval = true)
    private List<Lesson> lessons = new ArrayList<>();

    @OneToMany(mappedBy = "debrief", cascade = CascadeType.ALL,
            fetch = FetchType.LAZY, orphanRemoval = true)
    private List<Task> tasks = new ArrayList<>();
}

File: DebriefRepository/src/main/java/org/example/debriefrepository/entities/Debrief.kt
package org.example.debriefrepository.entities

import jakarta.persistence.*
import java.time.LocalDate

@Entity
@Table(name = "debrief", schema = "debrief_mgmt")
open class Debrief {
    @Id
    @Column(name = "id", nullable = false)
    open var id: Long? = null

    @Column(name = "content", nullable = false, length = Integer.MAX_VALUE)
    open var content: String? = null

    @Column(name = "date", nullable = false)
    open var date: LocalDate? = null

    @ManyToMany
    @JoinTable(
        joinColumns = [JoinColumn(name = "debrief_id")],
        inverseJoinColumns = [JoinColumn(name = "lesson_id")]
    )
    open var lessons: MutableSet<org.example.debriefrepository.entities.Lesson> = mutableSetOf()

    @ManyToMany
    @JoinTable(
        joinColumns = [JoinColumn(name = "debrief_id")],
        inverseJoinColumns = [JoinColumn(name = "mission_id")]
    )
    open var missions: MutableSet<org.example.debriefrepository.entities.Mission> = mutableSetOf()

    @ManyToMany
    @JoinTable(
        name = "group_debriefs",
        joinColumns = [JoinColumn(name = "debrief_id")],
        inverseJoinColumns = [JoinColumn(name = "group_id")]
    )
    open var groups: MutableSet<org.example.debriefrepository.entities.Group> = mutableSetOf()

    @ManyToMany
    @JoinTable(
        name = "users_debriefs",
        joinColumns = [JoinColumn(name = "debrief_id")],
        inverseJoinColumns = [JoinColumn(name = "user_id")]
    )
    open var users: MutableSet<org.example.debriefrepository.entities.User> = mutableSetOf()
}